features/2025-07-10-offline-install-feature-spec.md

```diff
@@ -0,0 +1,407 @@
+
+The key changes from today's `install.sh` include:
+
+* Today, `rad-bicep` is installed via the `rad bicep download` command. This approach is problematic in that is is not obvious that Radius is downloading a new binary. The command also installs the `rad-bicep` binary in a different directory (`~/.rad/bin`) which is not ideal. The `install.sh` script is enhanced to install the `rad-bicep` binary side-by-side `rad` and the `rad bicep download` command is removed.
+* Today, the `bicepconfig.json` file is created by `rad init`. This is problematic since offline installs will not use `rad init`. Also, the contents of `bicepconfig.json` file created by `rad init` is hard-coded. With this change, the `install.sh` script will create the `bicepconfig.json` file based on user input (`--bicep-extension-oci-registry`, and `--custom-resource-type-extension-location`)
+
+### User Story 3 – Install Radius via CLI
+
+***As a platform engineer, I need to install Radius offline using the Radius CLI***
+
+The platform engineer installs Radius using the CLI specifying the private OCI registry and to skip installing Contour.
+
+```bash
+rad install kubernetes \
+  --set image.repository=<OCI_REGISTRY_URL>/<REPOSITORY>
+  --skip-contour-install
+```
+
+> [!NOTE]
+>
+>  `rad init` will not support offline installations.
+
+The `image.repository` is the private OCI registry where the Radius container images were stored in the previous step.
+
+The `--skip-contour-install` does not install Contour. Therefore the Gateway resource types will not function until extensibility is implemented.
+
+> [!CAUTION]
+>
+> `--skip-contour-install` is expected to be removed from Radius in the near future as part of the compute extensibility work. The design for the Gateway recipe no longer uses Contour and Contour is no longer installed by default.
+
+If a Helm mirror has not been configured via `helm repo add`, the `--chart` argument can be used.
+
+```bash
+rad install kubernetes \
+  --chart <PATH_TO_RADIUS_HELM_CHART_FROM_HELM_PULL> \
+  ...
+```
+
+### User Story 4 – Install Radius via Helm
+
+***As a platform engineer, I need to install Radius using Helm***
+
+The platform engineer installs Radius using Helm with similar settings.
+
+```bash
+helm upgrade radius radius/radius \
+  --install \
+  --create-namespace 
+  --namespace radius-system \
+  --version 0.48.0 \
+  --wait --timeout 15m0s \
+  --set image.repository=<OCI_REGISTRY_URL>/<REPOSITORY>
+```
+
+## Scenario 2 – Managing Terraform
+
+### User Story 5 – Install Terraform
+
+***As a platform engineer, I need to install and configure Terraform for Radius to use***
+
+The platform engineer deploys a Terraform resource. The Terraform resource replaced the `recipeConfig` properties in the Environment today and adds all other [terraformrc configuration options](https://developer.hashicorp.com/terraform/cli/config/config-file).
```

Comment:
Some context on how Terraform is installed and configured today in Radius would be helpful